src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -3058,7 +3090,35 @@ void WindowRideConstructionUpdateActiveElementsImpl()
         {
             _selectedTrackType = tileElement->AsTrack()->GetTrackType();
             if (TrackTypeHasSpeedSetting(tileElement->AsTrack()->GetTrackType()))
-                _currentBrakeSpeed2 = tileElement->AsTrack()->GetBrakeBoosterSpeed();
+            {
+                _currentBrakeSpeed = tileElement->AsTrack()->GetBrakeBoosterSpeed();
+#ifndef UNIFIED_BRAKE_SPEED_ENABLED
+                if (tileElement->AsTrack()->GetTrackType() == TrackElemType::Booster)
```

Comment:
This section forces the speed value to align with legacy speed values. Necessary for td6